src/reactviews/pages/QueryResult/table/plugins/contextMenu.plugin.ts

```diff
@@ -0,0 +1,136 @@
+                `<li data-action="copy" class="contextMenu">${locConstants.queryResult.copy}</li>` +
+                `<li data-action="copy-with-headers" class="contextMenu">${locConstants.queryResult.copyWithHeaders}</li>` +
+                `<li data-action="copy-headers" class="contextMenu">${locConstants.queryResult.copyHeaders}</li>` +
+                `</ul>`,
+        );
+        // Remove any existing context menus to avoid duplication
+        jQuery("#contextMenu").remove();
+
+        // Append the menu to the body and set its position
+        jQuery("body").append($contextMenu);
+
+        let cell = this.grid.getCellFromEvent(e);
+        $contextMenu
+            .data("row", cell.row)
+            .css("top", mouseEvent.pageY)
+            .css("left", mouseEvent.pageX)
+            .show();
+
+        jQuery("body").one("click", function () {
+            $contextMenu.hide();
+        });
+
+        $contextMenu.on("click", "li", async (event) => {
+            const action = jQuery(event.target).data("action");
+            await this.handleMenuAction(action);
+            $contextMenu.hide(); // Hide the menu after an action is clicked
+        });
+    }
+
+    private async handleMenuAction(action: string): Promise<void> {
+        let selectedRanges = this.grid.getSelectionModel().getSelectedRanges();
+        let selection = tryCombineSelectionsForResults(selectedRanges);
+        switch (action) {
+            case "select-all":
+                console.log("Select All action triggered");
+                const data = this.grid.getData() as HybridDataProvider<T>;
+                let selectionModel = this.grid.getSelectionModel();
+                selectionModel.setSelectedRanges([
+                    new Slick.Range(
+                        0,
+                        0,
+                        data.length - 1,
+                        this.grid.getColumns().length - 1,
+                    ),
+                ]);
+                break;
+            case "copy":
+                await this.webViewState.extensionRpc.call("copySelection", {
+                    uri: this.uri,
+                    batchId: this.resultSetSummary.batchId,
+                    resultId: this.resultSetSummary.id,
+                    selection: selection,
+                });
+
+                console.log("Copy action triggered");
+                break;
+            case "copy-with-headers":
+                await this.webViewState.extensionRpc.call("copyWithHeaders", {
+                    uri: this.uri,
+                    batchId: this.resultSetSummary.batchId,
+                    resultId: this.resultSetSummary.id,
+                    selection: selection,
+                });
+
+                console.log("Copy with Headers action triggered");
+                break;
+            case "copy-headers":
+                await this.webViewState.extensionRpc.call("copyAllHeaders", {
```

Comment:
oh I didn't realize this, I went with copy all headers because that's what we currently have in the extension.  Should I change it?

Reply:
Yeah, let's change it to just copying the headers, since that's what ADS does:

output: 
> `principal_id	schema_id	parent_object_id	type	type_desc`

![image](https://github.com/user-attachments/assets/71f4bd53-a307-453b-8d2d-55b1b1f7f70e)